frontend/app/services/SubscriptionService.scala

```diff
@@ -154,28 +155,51 @@ class SubscriptionService(val tierPlanRateIds: Map[ProductRatePlan, String],
 
   def memberTierFeatures(memberId: MemberId): Future[Seq[Rest.Feature]] = for {
     (_, subscription) <- accountWithLatestMembershipSubscription(memberId)
-    catalog  <- zuoraRestService.productCatalogSupplier.get
-  } yield subscription
-      .latestWhiteListedRatePlan(catalog.productIdsOfType(MembershipProductType)).toSeq
-      .flatMap(_.subscriptionProductFeatures)
+    ratePlanOpt <- currentRatePlan(subscription)
+  } yield {
+      val features = ratePlanOpt.toSeq.flatMap(_.subscriptionProductFeatures)
+      logger.debug(
+        s"Checking product features for member ${memberId.salesforceAccountId}." ++
+        s" Current product ${ratePlanOpt.map(_.productName)}, features: ${features.map(_.featureCode)}")
+      features
+    }
+
+  def currentRatePlan(subscription: Rest.Subscription): Future[Option[Rest.RatePlan]] = subscription.ratePlans match {
+    case onlyRatePlan :: Nil => Future.successful(Some(onlyRatePlan))
+    case multipleRatePlans => Timing.record(zuoraSoapService.metrics, "currentRatePlan-for-multipleRatePlans") {
+      for {
+        subscriptionStatus <- getSubscriptionStatus(subscription)
+        currentRatePlan <- getRatePlan(subscriptionStatus.currentVersion)
+      } yield {
+        logger.debug(s"Current ratePlan: $currentRatePlan, subscriptionStatus: $subscriptionStatus, multipleRatePlans: $multipleRatePlans")
+        multipleRatePlans.find(_.productRatePlanId == currentRatePlan.productRatePlanId)
```

Comment:
It's worth noting that this is a bit of hack: we're matching on the ProductRatePlanId (ie the ids used in the Product Catalog) rather than the RatePlanId (the ids that relate to a specific instance of a rate plan used by a customer).

The SOAP and Rest Zuora APIs appeared to return different RatePlanIds, preventing us from matching, so we've resorted to ProductRatePlanId - which should produce the same results for our purposes.